### PR TITLE
Test and document Java 22 compile/test support

### LINKED ISSUE #26245: Support compiling and testing Java 22
```[tasklist]
### Steps to complete
- [x] Add JDK to CI (separate PR, so the below is testable)
- [ ] Bump ASM to 9.6 (https://asm.ow2.io/versions.html)
- [ ] Update `AsmConstants`
- [ ] Remove unsupported warning from `JavaVersion`
- [ ] Add future entry to `JavaVersion` to maintain 2-year window of future version constants
- [ ] Update `JacocoCoverage` cutoffs if necessary
- [ ] Update compatibility table
- [ ] Update `GroovyCompileToolchainIntegrationTest#can compile source and run tests using Java #javaVersion for Groovy`, and the similar one for `ScalaCompileToolchainIntegrationTest`
- [ ] Update `JavaVersionSpec` and `JavaClassUtilTest`
- [ ] Update list of cross-compile JavaVersions in `JavaCrossCompilationIntegrationTest`
```

### PR DESCRIPTION
**Pull request by octylFractal**
Fixes #26245 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/core-runtime/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy [hunk 1]
```diff
@@ -60,7 +60,8 @@ class JavaVersionSpec extends Specification {
         JavaVersion.VERSION_23.toString() == "23"
         JavaVersion.VERSION_24.toString() == "24"
         JavaVersion.VERSION_25.toString() == "25"
-        JavaVersion.VERSION_HIGHER.toString() == "26"
+        JavaVersion.VERSION_26.toString() == "25"
```
Comment (mlopatkin) [inline platforms/core-runtime/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy:63]: ❓ should this be `== "26"`?

Comment (octylFractal) [reply platforms/core-runtime/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy:63]: Ah, yes.

**Review by mlopatkin — APPROVED** [review]

**Review by big-guy — CHANGES_REQUESTED** [review]
Please add an entry in the release notes for Java 22 support.

**Comment by octylFractal** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

File: platforms/documentation/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -52,6 +52,15 @@ Example:
 ADD RELEASE FEATURES BELOW
 vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
 
+<a name="java-22"></a>
+### Support for building projects with Java 22
+
+Gradle now supports [using Java 22](userguide/compatibility.html#java) for compiling, testing, and starting other Java programs.
+This can be accomplished using [toolchains](userguide/toolchains.html).
+
+Currently, you cannot run Gradle on Java 22 because Groovy lacks support for JDK 22.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/release/notes.md:61]: Even for Kotlin DSL?

Comment (octylFractal) [reply platforms/documentation/docs/src/docs/release/notes.md:61]: Kotlin DSL is... more complicated. Simple buildscripts might work, but projects that apply `kotlin-dsl` require manual configuration. I don't know if it's worth specifying that difference here.

**Review by oleg-nenashev — APPROVED** [review]
+1, and thanks for the comment w.r.t Kotlin

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**